impl/sql/mutationstorage/queue_test.go

```diff
@@ -131,48 +131,48 @@ func TestQueueSendBatch(t *testing.T) {
 
 	for _, tc := range []struct {
 		description string
-		updates     []*pb.EntryUpdate
+		updates     [][]*pb.EntryUpdate
 		batchSize   int32
 	}{
 		{
-			description: "read half",
-			updates: []*pb.EntryUpdate{
-				genUpdate(1),
-				genUpdate(2),
-				genUpdate(3),
-			},
-			batchSize: 3,
-		},
-		{
-			description: "read rest",
-			updates: []*pb.EntryUpdate{
-				genUpdate(4),
-				genUpdate(5),
+			description: "read half and rest",
+			updates: [][]*pb.EntryUpdate{
+				{
+					genUpdate(1),
+					genUpdate(2),
+					genUpdate(3),
+				},
+				{
+					genUpdate(4),
+					genUpdate(5),
+				},
 			},
 			batchSize: 3,
 		},
 		{
 			description: "empty queue",
-			updates:     nil,
+			updates:     [][]*pb.EntryUpdate{{}},
 			batchSize:   10,
 		},
 	} {
-		wg := new(sync.WaitGroup)
-		wg.Add(1)
+		var wg sync.WaitGroup
+		wg.Add(len(tc.updates))
+		var i int
```

Comment:
Are you sure this should be init'd to zero outside the t.Run loop? It looks dubious, and maybe fragile if tests get omitted via a test filter.

I'd expect var i int (or just i :=0) to be just before r, ok := m.NewReceiver within the t.Run loop.

Reply:
woops, yes, this should be inside t.Run